src/components/rules-creation-page/steps/IncludeExcludeSelectionStep.tsx

```diff
@@ -0,0 +1,236 @@
+    "organisationUnits.organisationUnitGroups": "Organisation unit groups of organisation units",
+    organisationUnitGroups: "Organisation unit groups",
+    "organisationUnitGroups.attributes": "Attributes of organisation unit groups",
+    "organisationUnitGroups.organisationUnits": "Organisation units in organisation unit groups",
+    "organisationUnitGroups.organisationUnits.attributes":
+        "Attributes of organisation units in organisation unit groups",
+    "organisationUnitGroups.organisationUnits.organisationUnitGroups":
+        "Organisation unit groups of organisation units in organisation unit groups",
+    "organisationUnitGroups.organisationUnitGroupSets":
+        "Organisation unit group sets of organisation unit groups",
+    "organisationUnitGroups.organisationUnitGroupSets.attributes":
+        "Attributes of organisation unit group sets in organisation unit groups",
+    organisationUnitGroupSets: "Organisation unit group sets",
+    "organisationUnitGroupSets.attributes": "Attributes of organisation unit group sets",
+    users: "Users",
+    validationRules: "Validation rules",
+    "validationRules.attributes": "Attributes of validation rules",
+    "validationRules.validationRuleGroups": "Validation rule groups of validation rules",
+    validationRuleGroups: "Validation rule groups",
+    "validationRuleGroups.attributes": "Attributes of validation rule groups",
+};
+
+const IncludeExcludeSelectionStep: React.FC<IncludeExcludeSelectionStepProps> = ({
+    d2,
+    syncRule,
+    onChange,
+}) => {
+    const [modelSelectItems, setModelSelectItems] = useState<ModelSelectItem[]>([]);
+    const [models, setModels] = useState<typeof D2Model[]>([]);
+    const [selectedType, setSelectedType] = useState<string | undefined>();
+
+    useEffect(() => {
+        getMetadata(getBaseUrl(d2), syncRule.metadataIds, "id,name").then((metadata: any) => {
+            const models = _.keys(metadata).map((type: string) => {
+                return d2ModelFactory(d2, type);
+            });
+
+            const parseModels = (models: typeof D2Model[]) =>
+                models
+                    .map((model: typeof D2Model) => model.getD2Model(d2))
+                    .map((model: ModelDefinition) => ({
+                        name: model.displayName,
+                        id: model.name,
+                    }));
+
+            setModels(models);
+            setModelSelectItems(parseModels(models));
+        });
+
+        console.log(syncRule.metadataExcludeIncludeRules);
+    }, [d2, syncRule]);
+
+    const changeUseDefaultIncludeExclude = (useDefault: boolean) => {
+        onChange(
+            useDefault
+                ? syncRule.markToUseDefaultIncludeExclude()
+                : syncRule.markToNotUseDefaultIncludeExclude(models)
+        );
+    };
+
+    const changeModelName = (event: any) => {
+        setSelectedType(event.target.value);
```

Comment:
In ``development-malaria`` we now have ``onValueChange`` like the ``Toggle`` component and already returns a ``string``,

Reply:
ok!